Libplanet/Net/ITransport.cs

```diff
@@ -1,49 +1,176 @@
+        /// </summary>
+        [Pure]
         Peer AsPeer { get; }
 
+        /// <summary>
+        /// List of all <see cref="Peer"/>s have in routing table.
+        /// </summary>
+        [Pure]
         IEnumerable<BoundPeer> Peers { get; }
 
+        /// <summary>
+        /// The <see cref="DateTimeOffset"/> of the last message was received.
+        /// </summary>
+        [Pure]
         DateTimeOffset? LastMessageTimestamp { get; }
 
+        /// <summary>
+        /// Setup transport layer.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// A cancellation token used to propagate notification that this
+        /// operation should be canceled.</param>
+        /// <returns>An awaitable task without value.</returns>
         Task StartAsync(CancellationToken cancellationToken = default(CancellationToken));
 
+        /// <summary>
+        /// Start running transport layer. To <see cref="RunAsync"/>, you should call
+        /// <see cref="StartAsync"/> first.
+        /// </summary>
+        /// <param name="cancellationToken">
+        /// A cancellation token used to propagate notification that this
+        /// operation should be canceled.</param>
+        /// <returns>An awaitable task without value.</returns>
         Task RunAsync(CancellationToken cancellationToken = default(CancellationToken));
 
+        /// <summary>
+        /// Stop running transport layer.
+        /// </summary>
+        /// <param name="waitFor">The <see cref="TimeSpan"/> of delay
+        /// before actual stopping.</param>
+        /// <param name="cancellationToken">
+        /// A cancellation token used to propagate notification that this
+        /// operation should be canceled.</param>
+        /// <returns>An awaitable task without value.</returns>
         Task StopAsync(
             TimeSpan waitFor,
             CancellationToken cancellationToken = default(CancellationToken));
 
+        /// <summary>
+        /// Adds a <see cref="EventHandler"/> invoked when a message that is not
+        /// a reply is received. To handle reply, please use <see cref=
+        /// "SendMessageWithReplyAsync(BoundPeer,Message,TimeSpan?,CancellationToken)"/>.
+        /// </summary>
+        /// <param name="eventHandler">A handler of event when a message
+        /// that is not a reply is received.</param>
+        void AddEventHandler(EventHandler<Message> eventHandler);
+
+        /// <summary>
+        /// Removes a <see cref="EventHandler"/> registered via <see cref="AddEventHandler"/>.
+        /// </summary>
+        /// <param name="eventHandler">A handler of event when a message
+        /// that is not a reply is received.</param>
+        void RemoveEventHandler(EventHandler<Message> eventHandler);
+
+        /// <summary>
+        /// Removes all <see cref="EventHandler"/> registered via <see cref="AddEventHandler"/>.
+        /// </summary>
+        void RemoveAllEventHandlers();
```

Comment:
What benefits do these methods to add/remove event handlers compared to adding a public event like `ProcessMessageHandler` to the interface?

Reply:
I did not know an event can be interface member. Thanks for the comment!